File: Common/Actor/src/common/actor/LevelElement.java
package common.actor;

abstract public class LevelElement extends Actor {

    public LevelElement( final double x, final double y, final String... pictureFilePaths ) {
        super( x, y, 0, pictureFilePaths );
    }

    protected LevelElement( double x, double y, String picturePath, double scale ) {
        super( x, y, scale, picturePath );
        this.scaleImage( scale );
    }
}

File: Common/Actor/src/common/actor/LevelElement.kt
package common.actor

abstract class LevelElement : Actor {

    protected constructor(x: Double,
                y: Double,
                pictureFilePaths: Array<String>,
                delay: Int = 0)
            : super(x = x, y = y, delay = delay, pictureFilePaths = pictureFilePaths)

    protected constructor(x: Double,
                          y: Double,
                          picturePath: String,
                          scale: Double) : super(picturePath, x, y) {
        scaleImage(scale)
    }
}
